src/object/srv_ec_aggregate.c

```diff
@@ -2436,18 +2489,36 @@ ds_obj_ec_aggregate(struct ds_cont_child *cont, daos_epoch_range_t *epr,
 
 	dtx_end(&dth, cont, rc);
 
-	if (daos_handle_is_valid(agg_param.ap_agg_entry.ae_obj_hdl))
-		dsc_obj_close(agg_param.ap_agg_entry.ae_obj_hdl);
+	if (daos_handle_is_valid(ec_agg_param->ap_agg_entry.ae_obj_hdl))
+		dsc_obj_close(ec_agg_param->ap_agg_entry.ae_obj_hdl);
+
+	if (rc == 0)
+		cont->sc_ec_agg_eph = max(cont->sc_ec_agg_eph, epr->epr_hi);
 
-	if (rc == 0 && is_current)
-		cont->sc_ec_agg_eph = epr->epr_hi;
-out_close:
-	dsc_cont_close(ph, agg_param.ap_pool_info.api_cont_hdl);
-out:
-	daos_prop_free(agg_param.ap_prop);
-	ABT_eventual_free(&agg_param.ap_pool_info.api_eventual);
-	d_sgl_fini(&agg_param.ap_agg_entry.ae_sgl, true);
-	if (daos_handle_is_valid(ph))
-		dsc_pool_close(ph);
 	return rc;
 }
+
+static uint64_t
+cont_ec_agg_start_eph_get(struct ds_cont_child *cont)
+{
+	return cont->sc_ec_agg_eph;
+}
+
+void
+ds_obj_ec_aggregate(void *arg)
+{
```

Comment:
minor, seems unnecessary to define ds_obj_ec_aggregate(), just move the func body to cont_ec_agg_ult() is enough

Reply:
then ec_agg_param_init/fini() needs to be exported to container module. anyway, this is trying to put all EC aggregate related stuff in one file.